src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -365,10 +366,10 @@ private static Expression AddResponseWritingToMethodCall(Expression methodCall,
                     // ExecuteTask<T>(action(..), httpContext);
                     else if (typeArg == typeof(string))
                     {
-                        return Expression.Call(
-                            ExecuteTaskOfStringMethod,
-                            methodCall,
-                            HttpContextExpr);
+                        var conditionalContentTypeExpr = HandleNullContentType();
```

Comment:
Nit: I don't see the benefit of `conditionalContentTypeExpr` local var. I'd just call the method where it's used.